### DISCUSSION TITLE
How to build tensorflow-compression package for aarch64?

### DISCUSSION THREAD

**Discussion by ganeshShindeQ** [tensorflow/compression#168] [Q&A]
I tried to follow steps 
I am using t2a machine with debian and used following steps to build tensorflow_compression
1)  git clone https://github.com/tensorflow/compression.git /tensorflow_compression &&
2)  cd /tensorflow_compression &&
3)  python -m pip install -U pip setuptools wheel &&
4)  python -m pip install -r requirements.txt &&
5)   bazel build -c opt --copt=-mavx  :build_pip_pkg
 
Got the following error 

Error: dd48f66785b37a83b2ce2c612bd47a6/external/bazel_tools/platforms/BUILD:84:6: in alias rule @bazel_tools//platforms:linux: Constraints from @bazel_tools//platforms have been removed. Please use constraints from @platforms repository embedded in Bazel, or preferably declare dependency on https://github.com/bazelbuild/platforms. See https://github.com/bazelbuild/bazel/issues/8622 for details

**Comment by jonaballe** [comment]
@ssjhv any ideas on this one?

**Comment by ssjhv** [comment]
I don't have experience or a machine for aarch64. But I suspect it is one
of the Bazel repos that TF depends on, because the only thing we use from
bazel_tools is http_archive.

This issue on highway may be a hint:
https://github.com/google/highway/pull/1094
According to that issue, Bazel 6 breaks certain deprecated features and
they could fix it by using a newer googletest commit. It looks like it is a
recent change, and TF on aarch64 or its dependencies on aarch64 may have a
similar issue.

In addition to the bazel error, there could be other issues as well. For
example, tensorflow seems to have a separate pypi name for aarch64:
https://pypi.org/project/tensorflow-aarch64/ so our requirements.txt may
need modification.
